main.cc

```diff
@@ -347,6 +349,44 @@ class sighup_handler {
     }
 };
 
+class sigquit_handler {
+    semaphore _sem;
+    gate _gate;
+    distributed<replica::database>& _db;
+    bool _stopped = false;
+
+    future<> dump_diagnostics() {
+        if (_stopped) {
+            return make_ready_future<>();
+        }
+        return get_units(_sem, 1).then([this] (auto units) {
+            return _db.invoke_on_all([] (replica::database& db) -> future<> {
+                diaglog.info("Diagnostics dump requested via SIGQUIT:\n{}\nIncrease logger level for extra diagnostics.", memory::generate_memory_diagnostics_report());
+                if (diaglog.level() <= log_level::info) {
+                    co_return;
+                }
+                co_await db.foreach_reader_concurrency_semaphore([] (reader_concurrency_semaphore& semaphore) {
+                    diaglog.debug("Extra diagnostics dump requested via SIGQUIT:\n{}", semaphore.dump_diagnostics());
+                    return make_ready_future<>();
```

Comment:
I would say that the reader concurrency semaphore dumps are more high-level that small pools etc.

Reply:
I thought it might be too verbose. The user can tweak the log level to get more.

Reply:
It may be too verbose, but it can save a round trip asking for diagnostics, realizing there isn't enough there, some back-and-forth to enable the log level, another attempt. This is already requested by the user, it's not some periodic log no one asked for.

Reply:
ok, I'll make it all INFO

Reply:
All diagnostics are dumped with INFO in v2.